### DISCUSSION TITLE
[Help]: Is it possible to make multiple row slider?

### DISCUSSION THREAD

**Discussion by romanslonov** [davidjerleke/embla-carousel#863] [Help] [answered]
### Summary

Hello, first of all, thank you a lot for your hard work on this lib, it's super awesome!

I am playing around to build the multiple row carousel, but have no luck.

![carousel_grid](https://github.com/davidjerleke/embla-carousel/assets/10784316/ba376995-2007-47c4-b8c1-ccbb4603435d)

Any idea on how to implement this?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [X] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by meirroth** [comment]
Hey @romanslonov, you can initiate as many carousels as you'd like, would this work for you? 
https://stackblitz.com/edit/embla-carousel-nuxt-863?file=app.vue

**Reply by romanslonov** [reply]
No, I just need one carousel with 2 rows

**Reply by meirroth** [reply]
@romanslonov Just trying to understand your use case, why doesn't two carousel instances work?

**Reply by meirroth** [reply]
Ah, I see, you want slides to wrap around?

**Reply by davidjerleke** [reply, marked as answer]
@romanslonov you might want to read up on this discussion which is related to this I think:

- https://github.com/davidjerleke/embla-carousel/discussions/816

You checked both the React and Vue package. Which one are you using?

---

Thanks @meirroth for helping out!

**Reply by romanslonov** [reply]
@davidjerleke Thank you a lot! I don't know how I couldn't find this discussion 🤯 My initial take was same as method 1, but thought there are better options. So I ended up with this:

```ts
const items = computed(() => {
  if (!props.data) return undefined;

  return props.data.reduce((acc, curr, index) => {
    if (index % 2 === 0) {
      acc.push([curr]);
    } else {
      acc[acc.length - 1].push(curr);
    }
    return acc;
  }, []);
});
```

```vue
<CarouselContent>
  <CarouselItem
    v-for="item in items"
    class="basis-4/5 min-[460px]:basis-[65%] min-[580px]:basis-[45%] md:basis-2/5 lg:basis-1/4"
  >
    <div class="space-y-4">
      <Card
        v-for="card in item"
        :title="card.title"
        :image="card.image"
      />
    </div>
  </CarouselItem>
</CarouselContent>
```

**Reply by davidjerleke** [reply]
@romanslonov I’m glad you got it sorted 🎉.
